build/builder.go

```diff
@@ -942,6 +945,40 @@ func sortDocuments(todo []*zoekt.Document) {
 	}
 }
 
+// sortDocuments2 sorts zoekt.Document according to their Scores. In general,
+// documents can have a nil score vector if the document to be indexed was
+// introduced after the ranking took place. A nil score vector translates to the
+// lowest possible rank.
+func sortDocuments2(rs []*zoekt.Document) {
```

Comment:
my spidey senses suggest adding a quick test here to ensure we don't ever panic, even if we have surprising invariants on the input.